homework-submission/week5-homework/src/components/TodoItem.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import Deadline from "./Deadline";
+class TodoItem extends React.Component {
+  state = {
+    isEditing: false
+  };
+  OnEdit = () => {
+    this.setState({ isEditing: true });
+  };
+  onCancel = () => {
+    this.setState({ isEditing: false });
+  };
+  onUpdateClick = e => {
+    e.preventDefault();
+    let id = this.props._id;
+    let newTodo = this.editedTodo.value;
+    this.props.onUpdate(id, newTodo);
+    this.setState({ isEditing: false });
+  };
+  renderButtons() {
+    if (this.state.isEditing) {
+      return (
+        <td className="actionButton todos-td">
+          <button onClick={this.onUpdateClick}> Update </button>
+          <button onClick={this.onCancel}>Cancel </button>
+        </td>
+      );
+    }
+    return (
+      <td className="actionButton todos-td">
+        <button onClick={this.OnEdit}> Edit </button>
+        <button onClick={this.props.onRemove}>Remove </button>
+      </td>
+    );
+  }
+  renderTodos() {
+    let done = this.props.done;
+    let listStyle = {
+      textDecoration: "line-through"
+    };
+    if (this.state.isEditing) {
+      return (
+        <td className="todos-td">
+          <form onSubmit={this.onUpdateClick}>
+            <input
+              defaultValue={this.props.description}
+              ref={el => (this.editedTodo = el)}
+              className="editing-input"
+            />
+          </form>
+        </td>
+      );
+    }
+    return (
+      <td style={done ? listStyle : {}} className="todos-td">
```

Comment:
Inline styling is always taking it's place, but in a real world applications you will end up with a very confusing part that is why the elements are not styled as expected...
and the answer is:  inline styling is takes it's affect.

I read this very interested [article](https://codeburst.io/4-four-ways-to-style-react-components-ac6f323da822) about React styling ways, you have any interest?